doc/source/ray-air/examples/torch_image_example.md

```diff
@@ -0,0 +1,267 @@
+
+    train_dataset_shard: torch.utils.data.Dataset = train.get_dataset_shard("train").to_torch(
+        feature_columns=["image"],
+        label_column="label",
+        batch_size=config["batch_size"],
+        unsqueeze_feature_tensors=False,
+        unsqueeze_label_tensor=False
+    )
+
+    for epoch in range(2):
+        running_loss = 0.0
+        for i, data in enumerate(train_dataset_shard):
+            # get the inputs; data is a list of [inputs, labels]
+            inputs, labels = data
+
+            # zero the parameter gradients
+            optimizer.zero_grad()
+
+            # forward + backward + optimize
+            outputs = model(inputs)
+            loss = criterion(outputs, labels)
+            loss.backward()
+            optimizer.step()
+
+            # print statistics
+            running_loss += loss.item()
+            if i % 2000 == 1999:  # print every 2000 mini-batches
+                print(f"[{epoch + 1}, {i + 1:5d}] loss: {running_loss / 2000:.3f}")
+                running_loss = 0.0
+
+        train.save_checkpoint(model=model.module.state_dict())
+```
+
+Finally, we can train our model. This should take a few minutes to run.
+
+```{code-cell} python3
+from ray.ml.train.integrations.torch import TorchTrainer
+
+trainer = TorchTrainer(
+    train_loop_per_worker=train_loop_per_worker,
+    train_loop_config={"batch_size": 2},
+    datasets={"train": train_dataset},
+    scaling_config={"num_workers": 2}
+)
+result = trainer.fit()
+latest_checkpoint = result.checkpoint
+```
+
+To scale your training script, create a [Ray Cluster](deployment-guide) and increase the number of workers. If your cluster contains GPUs, add `"use_gpu": True` to your scaling config.
+
+```{code-block} python
+scaling_config={"num_workers": 8, "use_gpu": True}
+```
+
+## Test the network on the test data
+
+Let's see how our model performs.
+
+To classify images in the test dataset, we'll need to create a {py:class}`Predictor <ray.ml.predictor.Predictor>`.
+
+{py:class}`Predictors <ray.ml.predictor.Predictor>` load data from checkpoints and efficiently perform inference. In contrast to {py:class}`TorchPredictor <ray.ml.predictors.integrations.torch.TorchPredictor>`, which performs inference on a single batch, {py:class}`BatchPredictor <ray.ml.batch_predictor.BatchPredictor>` performs inference on an entire dataset. Because we want to classify all of the images in the test dataset, we'll use a {py:class}`BatchPredictor <ray.ml.batch_predictor.BatchPredictor>`.
+
+```{code-cell} python3
+from ray.ml.predictors.integrations.torch import TorchPredictor
+from ray.ml.batch_predictor import BatchPredictor
+
+batch_predictor = BatchPredictor.from_checkpoint(
+    checkpoint=latest_checkpoint,
+    predictor_cls=TorchPredictor,
+    model=Net(),
+)
+
+outputs: ray.data.Dataset = batch_predictor.predict(
+    data=test_dataset, feature_columns=["image"], unsqueeze=False
+)
+
+outputs.show(1)
+```
+
+```{code-cell} python3
```

Comment:
Can you explain what you're doing in this cell?

Reply:
Fixed